llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -705,19 +828,20 @@ Instruction *InstCombinerImpl::foldFMulReassoc(BinaryOperator &I) {
   // has the necessary (reassoc) fast-math-flags.
   if (I.hasNoSignedZeros() &&
       match(Op0, (m_FDiv(m_SpecificFP(1.0), m_Value(Y)))) &&
-      match(Y, m_Sqrt(m_Value(X))) && Op1 == X)
+      match(Y, m_Sqrt(m_Value(X))) && Op1 == X && !delayFMulSqrtTransform(Op0))
```

Comment:
Why are these necessary? I'd expect the fdiv to get visited before the fmuls. Does your motivating case require them?

Also note that this kind of check is fundamentally unreliable due to multiple InstCombine runs. If the first InstCombine run does this transform and the second one would potentially be able to do the sqrt/div transform, then this check won't help.

Reply:
yes, fmul is getting visited first before fdiv for my motivating example and hence the check.

Reply:
Is it possible to make the transform work on the IR after the fmul fold instead of the one before?

Reply:
yes(by pattern matching fdiv instead of fmul ) unless that fmul fold is transformed again before this transform kicks in. Let me try